src/zenml/steps/step_decorator.py

```diff
@@ -146,6 +148,7 @@ def inner_decorator(func: F) -> Type[BaseStep]:
                     PARAM_OUTPUT_MATERIALIZERS: output_materializers,
                     PARAM_SETTINGS: settings,
                     PARAM_EXTRA_OPTIONS: extra,
+                    PARAM_SOURCE_CODE: inspect.getsource(func),
```

Comment:
Do we want to encode this to compress it / make safer?

Reply:
Good idea, added `zlib` encoding for the `zenml.config.Step` object in general.

Reply:
Removed encoding/decoding again following reviewer consensus